cloudify_types/cloudify_types/component/component.py

```diff
@@ -0,0 +1,511 @@
+                timeout=self.timeout,
+                pollster_args=pollster_args,
+                expected_result=False)
+
+        ctx.logger.info("Little wait internal cleanup services.")
+        time.sleep(POLLING_INTERVAL)
+        ctx.logger.info("Wait for stop all system workflows.")
+        pollster_args = \
+            dict(_client=self.client)
+        poll_with_timeout(
+            dep_system_workflows_finished,
+            timeout=self.timeout,
+            pollster_args=pollster_args,
+            expected_result=True)
+
+        if not self.blueprint.get(EXTERNAL_RESOURCE):
+            ctx.logger.info("Delete blueprint {0}."
+                            .format(self.blueprint_id))
+            client_args = dict(blueprint_id=self.blueprint_id)
+            self.dp_get_client_response('blueprints', BP_DELETE, client_args)
+
+        self._delete_plugins()
+        self._delete_secrets()
+        self._delete_properties()
+
+        return poll_result
+
+    def execute_workflow(self):
+        if 'executions' not in ctx.instance.runtime_properties.keys():
+            ctx.instance.runtime_properties['executions'] = dict()
+
+        update_attributes('executions', 'workflow_id', self.workflow_id)
+
+        # Wait for the deployment to finish any executions
+        pollster_args = \
+            dict(_client=self.client,
+                 _check_all_in_deployment=self.deployment_id)
+
+        if not poll_with_timeout(dep_system_workflows_finished,
+                                 timeout=self.timeout,
+                                 pollster_args=pollster_args,
+                                 expected_result=True):
+            return ctx.operation.retry(
+                'The deployment is not ready for execution.')
+
+        # we must to run some execution
+        if ((self.deployment.get(EXTERNAL_RESOURCE) and self.reexecute)
+                or not self.deployment.get(EXTERNAL_RESOURCE)):
+            execution_args = self.config.get('executions_start_args', {})
+            client_args = \
+                dict(deployment_id=self.deployment_id,
+                     workflow_id=self.workflow_id,
+                     **execution_args)
+            response = self.dp_get_client_response('executions',
+                                                   EXEC_START, client_args)
+
+            # Set the execution_id for the last execution process created
+            self.execution_id = response['id']
+            ctx.logger.debug('Executions start response: {0}'.format(response))
+
+            # Poll for execution success.
+            if not self.verify_execution_successful():
+                ctx.logger.error('Deployment error.')
+
+            ctx.logger.debug('Polling execution succeeded')
+
+        type_hierarchy = ctx.node.type_hierarchy
+
+        if DEP_TYPE in type_hierarchy:
+            ctx.logger.info('Start post execute component')
+            return self.post_execute_component()
```

Comment:
I think since we are only doing deployment we don't even need this. At least change it to:
```python
if DEP_TYPE not in ctx.node.type_hierarchy:
    return # or even raise
```
....maybe we should do that at the beginning. Or do we even care about someone hijacking this fn?

Reply:
It's in the impl of Component, so I agree of just removing it